clang/include/clang/Basic/AttrDocs.td

```diff
@@ -1994,6 +1994,23 @@ can be lowered.
   }];
 }
 
+def SYCLIntelBufferLocationAttrDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "kernel_args_buffer_location";
+  let Content = [{
+This attribute is implicitly added to OpenCL pointer kernel parameters generated
+from a SYCL kernel object. It lacks a spelling, as it is not intended to be used
+by the programmer.
+
+This attribute causes clang to generate metadata on the OpenCL kernel containing
+the number of kernel parameters. The metadata contains an integer that is set
+according to the values passed through the ``accessor`` property
+``buffer_location``. These values are mapped to the actual locations of the
+global buffers (such as DDR, QDR, etc) and applied to pointer kernel parameters.
+The default value is -1.
```

Comment:
Now that I see this, please clarify the last sentence here with the meaning/difference between -1 and no-metadata (and clarify that the default value for the metadata is -1)..  I realize it is pretty subtle, so I won't try to author it myself.